fubectl.source

```diff
@@ -1,30 +1,54 @@
 # helper functions
 alias _inline_fzf="fzf --multi --ansi -i -1 --height=50% --reverse -0 --header-lines=1 --inline-info --border"
 alias _inline_fzf_nh="fzf --multi --ansi -i -1 --height=50% --reverse -0 --inline-info --border"
+_isNamespaced(){
+    if [[ "$1" == "" ]]; then
+        echo "please specify a Kubernetes object!"
+        exit 1
+    fi
+    # ask cluster about not namespaced resources
+    if [ $(kubectl api-resources --no-headers --namespaced=false | awk '{print $1,$2,$4}' | sed 's/ /\n/g' | grep --count -i "^$1$")  -gt 0 ]; then
+        return 0
+   else
+        return 1
+    fi
```

Comment:
```
if x; then
  return 0
else
  return 1
fi
```

Should be equivalent with:
```
x
```
right? This code can be really simplified.

Reply:
No there are objects what can match twice. like nodes:
```
kubectl api-resources --namespaced=false | grep -i '^nodes'
nodes                             no                                          false        Node
nodes                                          metrics.k8s.io                 false        NodeMetrics
```
I wanted to make the if's at the `k` commands as clean as possible

Reply:
So my `if x` is your `if [ $(kubectl api-resources --no-headers --namespaced=false | awk '{print $1,$2,$4}' | sed 's/ /\n/g' | grep --count -i "^$1$")  -gt 0 ]`.
So your `x` is `[ $(kubectl api-resources --no-headers --namespaced=false | awk '{print $1,$2,$4}' | sed 's/ /\n/g' | grep --count -i "^$1$")  -gt 0 ]`.

I am just saying that you can remove the `if ` and the `; then` and also the whole:
```
  return 0
else
  return 1
fi
```
lines.

It will have the exact same effect. This is independent of `x` and has nothing to do with what you want to match or not.

Reply:
oooh, didn't know about this. Thx!

Reply:
done, take a look